app/src/main/java/io/github/droidkaigi/confsched2017/view/activity/SplashActivity.java

```diff
@@ -35,13 +53,34 @@ protected void onCreate(Bundle savedInstanceState) {
     @Override
     protected void onStart() {
         super.onStart();
-        handler.postDelayed(moveActivityRunnable, 3000);
+        loadSessionsForCache();
     }
 
     @Override
     protected void onStop() {
         super.onStop();
-        handler.removeCallbacks(moveActivityRunnable);
+        compositeDisposable.dispose();
+    }
+
+    private void loadSessionsForCache() {
+        Disposable disposable = Single.zip(sessionsRepository.findAll(Locale.getDefault()),
+                mySessionsRepository.findAll(),
+                delaySingleSource(),
+                (sessions, mySessions, obj) -> Observable.empty())
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .doFinally(() -> {
+                    if (isFinishing()) return;
+                    startActivity(MainActivity.createIntent(SplashActivity.this));
+                })
+                .subscribe();
+        compositeDisposable.add(disposable);
+    }
+
+    private Single<Object> delaySingleSource() {
+        return Single.create(e -> e.onSuccess(Observable.empty()))
+                .observeOn(AndroidSchedulers.mainThread())
+                .delay(MINIMUM_LOADING_TIME, TimeUnit.MILLISECONDS);
```

Comment:
If you use Single.timer, all you need is to rewrite loadSessionsForCache as below.

```
Single.zip(sessionsRepository.findAll(Locale.getDefault()),
                mySessionsRepository.findAll(),
                Single.timer(MINIMUM_LOADING_TIME, TimeUnit.MILLISECONDS),
                (sessions, mySessions, obj) -> Observable.empty())
```